app/models/submission_response.rb

```diff
@@ -6,14 +6,15 @@ class SubmissionResponse < ApplicationRecord
 
   def name
     # self.public_send(Question::INPUT_TYPES_AND_STORAGE[question.input_type])
-    name = nil
-    [string_response, text_response, YAML.load(boolean_response.to_s), integer_response.to_s,
```

Comment:
I believe `YAML.load(boolean_response.to_s)` was equivalent to `boolean_response`. The changes below are mostly a refactoring to simplify but also drops the unnecessary double-conversion.